code/game/machinery/turrets/turrets.dm

```diff
@@ -0,0 +1,225 @@
+// Base type for rewritten turrets, designed to hopefully be cleaner and more fun to use and fight against.
+// Use the subtypes for 'real' turrets.
+
+/obj/machinery/turret
+	name = "don't use me"
```

Comment:
Something like `abstract turret` would probably be slightly clearer. You can also do something like `base_type = /obj/machinery/turret` (which is required for constructable machines anyway) and then in `Initialize()` do `if(base_type == type) PRINT_STACK_TRACE("Base abstract turret type instantiated, use a subtype instead!")`.